File: codility/php/OddOccurrencesInArray.php
<?
// Lesson 2. Arrays - OddOccurrencesInArray

function solution ($A)
{
    $answer = 0;

    sort($A);

    while (1) {
        if (count($A) == 1) {
            $answer = $A[0];
            break;
        } else {
            if ($A[0] !== $A[1]) {
                $answer = $A[0];
                break;
            }

            $A = array_slice($A, 2);
        }
    }

    return $answer;
}
/*
function solution($A) {
    $answer = 0;

    $A = array_count_values($A);
    foreach($A as $number => $count) {
        if ($count % 2) {
            $answer = $number;
            break;
        }
    }
    
    return $answer;
}
*/

File: codility/python/OddOccurrencesInArray.py
# Lesson 2. Arrays - OddOccurrencesInArray

def solution(A):
    answer = 0
    A.sort(reverse=True)

    while A:
        if len(A) == 1:
            answer = A[0]
            break

        num1 = A.pop()
        num2 = A.pop()
        if num1 != num2:
            answer = num1
            break

    return answer

answer = solution([5,1,3,2,5,3,4,4,2])
print(answer)

